components/eamxx/src/physics/dp/dp_functions.hpp

```diff
@@ -193,8 +198,45 @@ struct Functions
     const uview_1d<Spack>& t_update,   // temperature [m/s]
     const uview_2d<Spack>& q_update);  // tracer [vary]
 
-  KOKKOS_FUNCTION
-  static void iop_setinitial(const Int& nelemd, const uview_1d<element_t>& elem);
+  //---------------------------------------------------------
+  // Purpose: Set initial values from IOP files (where available)
+  //   when running SCM or DP-CRM.
+  //----------------------------------------------------------
+  static void iop_setinitial(
+    // Input arguments
+    const Int& plev,                        // number of vertical levels
+    const Int& pcnst,                       // number of advected constituents including cloud water
+    const Int& nelemd,                      // number of elements per MPI task
+    const Int& np,                          // NP
+    const Int& nstep,                       // the timestep number
+    const bool& use_replay,                 // use e3sm generated forcing
```

Comment:
A lot of these params were coming from iop_data_mod, which has over 100 fields. I'm not sure what the best approach is here. We could make an iop_data_mod struct but that would require a lot more porting from f90. How would we initialize all that data?

Reply:
As I noted above, maybe having a dp_interface layer which collects all this stuff and bundles it into structs which we pass to these routines would be helpful. Then unit tests would just hardcode values for all these structs as needed, while the interface would collect it from elsewhere in the code (so no init needed).